lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -223,8 +223,8 @@ sub _stash_job_and_module_list {
 sub details {
     my ($self) = @_;
 
-    $self->_stash_job_and_module_list or return $self->reply->not_found;
-    $self->render('test/details');
+    my $job = $self->_stash_job_and_module_list or return $self->reply->not_found;
+    $self->render($job->should_show_autoinst_log ? 'test/autoinst_log_within_details' : 'test/details');
```

Comment:
That's neat actually. I saw the file at first wondering why you were splitting that out.